src/scr_flush_file.c

```diff
@@ -40,6 +45,184 @@
 #error "globals.h accessed from tools"
 #endif
 
+/*
+ * Remove a particular dataset from the flush file.
+ */
+static void
+scr_flush_file_remove_dataset(int id, char* flush_file_path)
+{
+    kvtree* hash = kvtree_new();
+    kvtree_read_file(flush_file_path, hash);
+
+    kvtree_unset_kv_int(hash, SCR_FLUSH_KEY_DATASET, id);
+
+    /* write the hash back to the flush file */
+    kvtree_write_file(flush_file_path, hash);
+
+    /* delete the hash */
+    kvtree_delete(&hash);
+}
+
+/*
+ * Given a path to a prefix directory, the contents of the flush file,
+ * an id, and the path to the flush_file, generate the summary.scr file and
+ * update flush.scr to show that we're no longer flushing.
+ *
+ * Returns 0 on success.  Returns 1 if there's no corresponding dataset_id
+ * in the flush file.
+ */
+int write_summary_file(char* prefix, kvtree* flush_file, int dataset_id,
+  char* flush_file_path)
+{
+
+  /* define metadata directory for dataset */
+  char* summary_file = NULL;
+  int rc;
+  kvtree* flush_key_dataset;
+  kvtree* summary;
+
+  flush_key_dataset = kvtree_get_kv_int(flush_file, SCR_FLUSH_KEY_DATASET, dataset_id);
+  if (flush_key_dataset == NULL) {
+    scr_err("No flush file entry for dataset %d\n", dataset_id);
+    return 1; /* No entry for this dataset */
+  }
+
+  summary = kvtree_get(flush_key_dataset, SCR_FLUSH_KEY_DSETDESC);
+  asprintf(&summary_file, "%s/.scr/scr.dataset.%d/summary.scr", prefix, dataset_id);
+  rc = scr_flush_summary_file(summary, 1, summary_file);
+  scr_free(&summary_file);
+
+  /* All done flushing */
+  scr_flush_file_remove_dataset(dataset_id, flush_file_path);
+
+  return rc;
+}
+
+/*
+ * Given a path to a state_file, resume and finalize all transfers for all
+ * files in the state_file.
+ */
+void resume_transfer(char* state_file_path)
+{
+  int id;
+  int rc;
+  AXL_Init();
+
+  id = AXL_Create(AXL_XFER_STATE_FILE, "scr", state_file_path);
+  if (id < 0) {
+    scr_err("Error: AXL_Create() = %d\n", id);
```

Comment:
For ``scr_err`` calls, let's change them to be of this form:
```suggestion
    scr_err("Error: AXL_Create() = %d @ %s:%d", id, __FILE__, __LINE__);
```
The function adds a trailing newline, so that's not needed.  We also tack on the file and line number to every message, because that info helps in debugging problems reported by users.